server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageDependenciesModule.java

```diff
@@ -82,4 +85,15 @@ private ObjectStorageBlobsDAO buildObjectStore(ObjectStorageBlobConfiguration co
         throw new IllegalArgumentException("unknown provider " + configuration.getProvider());
     }
 
+    private Optional<PutBlobFunction> putBlob(BlobId.Factory blobIdFactory, ObjectStorageBlobConfiguration configuration) {
+        switch (configuration.getProvider()) {
+            case SWIFT:
+                return Optional.empty();
```

Comment:
Would not it be better to directly return a function here?

Reply:
Hard todo without breaking the `Supplier<BlobStore>` mechanism.